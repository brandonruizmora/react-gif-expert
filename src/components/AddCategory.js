import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleSubmit = function (e) {
        e.preventDefault();
        if (inputValue.trim() !== '') {
            setCategories(categories => [inputValue, ...categories]);
            setInputValue('');
        }
    }

    const handleInputChange = function (e) {
            setInputValue(e.target.value);
    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    );
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}

export default AddCategory;
