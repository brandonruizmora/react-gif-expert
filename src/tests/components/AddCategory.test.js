import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Pruebas en <AddCategory />', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    });

    test('Debe de mostarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe cambiar la caja de texto', () => {
        const inputBefore = wrapper.find('input');
        const event = { target: { value: 'Hola mundo'} };
        inputBefore.simulate('change', event);
        // console.log(inputBefore.html())
        const inputAfter = wrapper.find('input');
        // console.log(inputAfter.html())
        expect(inputAfter.prop('value')).toBe('Hola mundo');
    });

    test('No debe de postear la informacion con submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault() { } });
        expect(setCategories).not.toHaveBeenCalled();
    });

    test('Debe de llamar el setCategories y limpiar la caja de texto', () => {
        //Arrange
        const value = 'Goku';
        const event = { target: { value } };
        const eventSubmit = { preventDefault() { } };
        //Act
        wrapper.find('input').simulate('change', event );
        wrapper.find('form').simulate('submit', eventSubmit );
        //Assert
        expect( setCategories ).toHaveBeenCalled();
        expect( wrapper.find('input').prop('value') ).toBe('');
    })
    




});
