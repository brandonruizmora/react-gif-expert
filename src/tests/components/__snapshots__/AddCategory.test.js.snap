// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> Debe de mostarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    value=""
  />
</form>
`;
